core/coreapi/unixfs.go

```diff
@@ -55,11 +58,30 @@ func getOrCreateNilNode() (*core.IpfsNode, error) {
 // Add builds a merkledag node from a reader, adds it to the blockstore,
 // and returns the key representing that node.
 func (api *UnixfsAPI) Add(ctx context.Context, files files.Node, opts ...options.UnixfsAddOption) (path.Resolved, error) {
+	ctx, span := tracing.Span(ctx, "CoreAPI.UnixfsAPI", "Add")
+	defer span.End()
+
 	settings, prefix, err := options.UnixfsAddOptions(opts...)
 	if err != nil {
 		return nil, err
 	}
 
+	span.SetAttributes(
```

Comment:
Allocations can be avoided here (and elsewhere when attributes are used) by only setting them when the span is recording. e.g.
```go
if span.IsRecording() {
   // set stuffs
}
```

Reply:
Admittedly, this is tedious for some of the one-liners that start a span and set attributes together; I wonder if there is an alternative...

Reply:
I wonder if it's worth adding an `Attributes` method to some of these types like `settings,` for example.  It may be more appropriate for a follow-on.

Reply:
My thought process here was that we already know we have way bigger fish to fry in terms of memory allocations, so I wasn't too concerned with this.  Hopefully one day we get to the point where these matter...in the meantime, we can enjoy the simpler code